src/python/pants/core/util_rules/subprocess_environment.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import logging
+import os
+from dataclasses import dataclass
+from typing import List
+
+from pants.core.util_rules.pants_environment import PantsEnvironment
+from pants.engine.rules import _uncacheable_rule, collect_rules
+from pants.option.subsystem import Subsystem
+from pants.util.frozendict import FrozenDict
+
+logger = logging.getLogger(__name__)
+
+
+# Names of env vars that can be set on all subprocesses via config.
+SETTABLE_ENV_VARS = (
+    # Customarily used to control i18n settings.
+    "LANG",
+    "LC_CTYPE",
+    "LC_ALL",
+    # Customarily used to control proxy settings in various processes.
+    "http_proxy",
```

Comment:
very small nit: I would order these so that the uppercase and lowercase versions of each individual variable string are right next to each other, to make it easy to see that we specify both versions

Reply:
Eh, personally I disagree. I think this is fine either way though.